migrations/20220105195635_init.ts

```diff
@@ -0,0 +1,36 @@
+import { Knex } from 'knex';
+
+// Funny how this number ended up being the last changelog migration...
+const MAGIC_CHANGELOG_NUMBER = 42;
+
+export async function up(knex: Knex): Promise<void> {
+  // If table changelog is unavailable,
+  //  or if the last changelog migration row is less than < 42.
+
+  // Then complain and die.
+  try {
+    if (!await knex.schema.hasTable('changelog')) {
+      throw new Error('Old Migrations have never run, please apply those manually and then run `make migrate`.');
+    }
+
+    const result = await knex('changelog').select('id').orderBy('id', 'desc').limit(1);
+
+    if (result[0].id < MAGIC_CHANGELOG_NUMBER) {
+      throw new Error('Migrations must have reached `changelog` version 42 before applying Knex migrations, please apply those manually and then run `make migrate`.');
+    }
+
+  } catch (err: any) {
+    // Color the error red.
+    // eslint-disable-next-line no-console
+    console.error('\x1b[31m', err, '\x1b[37m');
+
+    process.exit(1);
```

Comment:
Instead of `exit,` you can also just `throw err` or just never catch this error at all (Knex should handle errors)

Reply:
Yeah let me remove the try catch all together